### ISSUE TITLE
[Choco Nuspec Checker] Incorrectly flags files as being packed. 

### ISSUE THREAD

**Issue by TheCakeIsNaOH** [bcurran3/ChocolateyPackages_CCR#194]
CNC is incorrectly saying that some files will be packed, even though they are not packed. This happens when there is the `<files>` section, but no `<file>` elements inside the `<files>` section. It says the files element is empty, and that the other files in the folder will be packed. That is incorrect, it is only if the `<files>` section/element is completely missing will it pack everything in the folder. 

Example of `anydesk`:
https://github.com/TheCakeIsNaOH/chocolatey-packages/blob/master/anydesk/anydesk.nuspec
https://chocolatey.org/packages/anydesk#files

Not a high priority or a big deal, just an inconsistency.

**Comment by bcurran3** [comment]
@TheCakeIsNaOH Hmmmmmmmmmmm.

You might have discovered a problem but I don't think you described the actual problem. :)

Take a look at your last two published anydesk packages you pushed:

https://chocolatey.org/packages/anydesk/6.0.8.0#files
https://chocolatey.org/packages/anydesk/6.0.7#files

There are no files in the package!

I think CNC needs to warn that if the files tag is there but empty, that no files will be included.


i.e. Chocolatey (choco pack) seems to behave differently when the files element is missing versus when the files element exists but is empty.  This seems like a bug we should bring up to the Chocolatey devs as the behavior is expected to be the same (IMHO).

**Comment by TheCakeIsNaOH** [comment]
> There are no files in the package!

Well, yes, this is intentional. The `anydesk` package depends on the `andesk.portable` package, which is the package that actually installs it. So the `anydesk` package does not have any extra files because there are none required.
https://chocolatey.org/packages/anydesk.portable#files
https://github.com/chocolatey/choco/wiki/ChocolateyFAQs#what-is-the-difference-between-packages-no-suffix-as-compared-to-install-portable

> I think CNC needs to warn that if the files tag is there but empty, that no files will be included.

Sounds great. 

> This seems like a bug we should bring up to the Chocolatey devs as the behavior is expected to be the same (IMHO).

We will probably have to agree to disagree on this one. If there are is no `files` element, it should pack everything, If there is a `files` element, it should pack whatever is in it. So if there is nothing inside the `files` element, it should pack no extra files, which is the current behavior. IMO, it should be documented (if it is not already documented), but not changed.

**Comment by bcurran3** [comment]
Hmmmmmm. <s>I'll have to go back and look at your packages again.</s> (Done) I didn't delve into the dependencies. At first look, it seemeded like there was  a problem with the packages as there were install scripts in the versions 6.0.6 and prior, but missing in 6.0.7+. Now I see you changed how the package works by making it a metapackage.

Intuitively, if the files element empty or missing should result in the same behavior; that how it works with all the other elements (off the top of my head). I'm not sure what you mean by "extra files" though. 

AFAIK the scenario is (or I used to believe):
files element - pack the specified files
no files element - pack everything in the current directory and deeper
blank files element - I THOUGHT it did the same as no files element, now I'm confused

I have not read the docs. I'm doing this off of what makes sense. I haven't even finished my first coffee or had breakfast yet....

**Comment by TheCakeIsNaOH** [comment]
Here is how I understand it currently works.

1. This packs the nuspec, and everything in the tools directory.
```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
  <metadata>
     ###Metadata here, removed for space
  </metadata>
  <files>
    <file src="tools\**" target="tools" />
  </files>
</package>
```


2. This packs everything in the current directory and all subdirectories
```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
  <metadata>
     ###Metadata here, removed for space
  </metadata>
</package>
```


3. This packs the nuspec only. Nothing else.
```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
  <metadata>
     ###Metadata here, removed for space
  </metadata>
  <files>
  </files>
</package>
```

It is case three I am talking about.

> I haven't even finished my first coffee or had breakfast yet....

I know the feeling.

**Comment by bcurran3** [comment]
OK. So I'll test and update appropriately.

**Comment by bcurran3** [comment]
Six months later, I tested! 
...and, of course, you're right.

I don't know how to check if the element exists or not versus the result of checking the value of the element without loading the whole file again and trying to match if the files element is there or not.  I didn't feel like adding yet another routine to load and check the file, so I went the lazy way and changed the message.

Some day in the distant future, I am open to more specific checking and splitting the two possibilities into two more specific message, but not today. :)

Updated in https://github.com/bcurran3/ChocolateyPackages/commit/692b7a176c23973c4cddca906535206d0d97524f
